### ISSUE TITLE
Adding a cli library

### ISSUE THREAD

**Issue by MariusVanDerWijden** [flashbots/mev-boost#640]
Would you be open to move the cli/flags handling from the basic-builtin handling to some library?

**Comment by ralexstokes** [comment]
I'd support this, esp for the right library -- is there one you have in mind?

I assume the intent is to simplify or harden the current flags infrastructure?

**Comment by MariusVanDerWijden** [comment]
Yes exactly, I would use `https://github.com/urfave/cli` which is the one we use in go-ethereum.
Theres a lot of dereferencing of flags going on which feels a bit unsafe

**Comment by Champ10q** [comment]
> Yes exactly, I would use `https://github.com/urfave/cli` which is the one we use in go-ethereum.
> Theres a lot of dereferencing of flags going on which feels a bit unsafe

**Comment by Champ10q** [comment]
I can implement this and will use urfave/cli (matches go-ethereum).

**Comment by Champ10q** [comment]
> Would you be open to move the cli/flags handling from the basic-builtin handling to some library?

**Comment by Champ10q** [comment]
Yes> Would you be open to move the cli/flags handling from the basic-builtin handling to some library?

**Comment by stlpruitt314** [comment]
Yes
